### ISSUE TITLE
New serial adapter

### ISSUE THREAD

**Issue by KostyaSha** [collin80/SavvyCAN#257]
Is there any guide on adding new serial adapter?
I see that serial  has enum types, but UI has no any settings and as soon as i create serial connection it sends  some gvret packets into adapter (attempt to split states and add buttons in UI #254 ).

**Comment by collin80** [comment]
Yeah, the serial connection is really just GVRET serial. It's hard coded to always try sending binary GVRET commands. If you want to add another serial adapter it would have to be a completely different class. It is a potentially good idea to abstract things such that there really is a base "serial" class and then GVRET and anything else sub-classes from that. The reason it is the way it is: I never meant to support anything other than GVRET originally. It was coded just for GVRET then other things were added over time. So, we're just getting outside the original scope.
